src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/validation/CalculateMixingFractions.java

```diff
@@ -0,0 +1,145 @@
+ * contributes to a pooled BAM.  We first estimate each sample's mixing fraction as the average alt fraction
+ * in the bam of singleton hets belonging to that sample, then normalize these initial estimates to sum to one.
+ *
+ * In the CRSP sensitivity validation, we have a bam derived from a pool of 5, 10, or 20 samples
+ * and a vcf of all known variants in those samples.  The pooled bam is a simulated tumor and
+ * the vcf of individual variants is our truth data.  We annotate the truth data with the estimated
+ * allele fractions of each variant in the pooled bam in order to bin our results by "tumor" allele fraction.
+ * To estimate allele fractions we need to know the sample mixing fractions as an intermediate step.
+ *
+ * One could use this tool for generating precise truth data any time multiple samples with known ground
+ * truth are combined to simulate tumors or any situation in which low allele fractions occur.
+ *
+ * Example usage:
+ * java -jat gatk.jar CalculateMixingFractions -V input.vcf -O output.table
+ *
+ * Created by David Benjamin on 1/30/17.
+ */
+@CommandLineProgramProperties(
+        summary = "Calculate proportions of different samples in a pooled bam by first estimating each sample's" +
+                " mixing fraction as the average alt fraction in the bam of singleton hets belonging to that sample, then" +
+                " normalizing these initial estimates to sum to one.",
+        oneLineSummary = "Calculate proportions of different samples in a pooled bam",
+        programGroup = VariantProgramGroup.class
+)
+public class CalculateMixingFractions extends VariantWalker {
+
+    @Argument(fullName= StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName=StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc="The output table of samplemixing fractions",
+            optional=false)
+    private final File outputFile = null;
+
+    private final Map<String, AltAndTotalReadCounts> sampleCounts = new HashMap<>();
+
+    @Override
+    public void onTraversalStart() {
+        getHeaderForVariants().getGenotypeSamples().forEach(s -> sampleCounts.put(s, new AltAndTotalReadCounts()));
+    }
+
+    @Override
+    public void apply(final VariantContext vc, final ReadsContext readsContext, final ReferenceContext refContext, final FeatureContext fc) {
+        if (!isBiallelicSingletonHetSnp(vc)) {
+            return;
+        }
+
+        final Optional<String> variantSample = StreamSupport.stream(vc.getGenotypes().spliterator(), false)
+                .filter(genotype -> genotype.isHet())
+                .map(genotype -> genotype.getSampleName())
+                .findFirst();
+
+        if (!variantSample.isPresent()) {
+            return;
+        }
+
+        final List<GATKRead> reads = new ArrayList<>();
+        final List<Integer> offsets = new ArrayList<>();
+
+        for (final GATKRead read : readsContext) {
+            if (read.failsVendorQualityCheck()) {
```

Comment:
I think filtering duplicates and MQ0 reads would be appropriate here too. What do you think?